### PR TITLE
feat: add pyproject.toml support (uv & poetry)

### PR DESCRIPTION
**Pull request by Strum355**
## Summary
- Add a new `DependencyProvider` for `pyproject.toml` files, supporting both PEP 621 (`[project.dependencies]`, `[project.optional-dependencies]`) and Poetry (`[tool.poetry.dependencies]`, `[tool.poetry.group.<name>.dependencies]`) formats
- Add VS Code settings for `uv` and `poetry` binary paths (`redHatDependencyAnalytics.uv.executable.path`, `redHatDependencyAnalytics.poetry.executable.path`), propagated to the JS client via `VSCEXT_TRUSTIFY_DA_UV_PATH` / `VSCEXT_TRUSTIFY_DA_POETRY_PATH` environment variables
- Register `pyproject.toml` in activation events, file handler, menus, and `rhda.ts` file type detection

Resolves: [TC-3852](https://redhat.atlassian.net/browse/TC-3852)

## Test plan
- [x] 13 unit tests covering PEP 621, Poetry, inline tables, dependency groups, extras, comments, position accuracy, edge cases
- [ ] Manual testing with `test-workspace/python-uv/pyproject.toml` (PEP 621 / hatchling)
- [ ] Manual testing with `test-workspace/python-poetry/pyproject.toml` (Poetry)
- [ ] Verify `uv.executable.path` and `poetry.executable.path` settings appear in VS Code settings UI
- [ ] Verify diagnostic underlines appear at correct positions for dependency versions

🤖 Generated with [Claude Code](https://claude.com/claude-code)

[TC-3852]: https://redhat.atlassian.net/browse/TC-3852?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/providers/pyproject.toml.ts [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+   */
+  private keyPathEquals(resolvedKey: (string | number)[], ...expected: string[]): boolean {
+    if (resolvedKey.length !== expected.length) { return false; }
+    return expected.every((seg, i) => resolvedKey[i] === seg);
+  }
+
+  /**
+   * Checks whether a resolved key starts with the given prefix segments.
+   */
+  private keyPathStartsWith(resolvedKey: (string | number)[], ...prefix: string[]): boolean {
```
Comment (ruromero) [inline src/providers/pyproject.toml.ts]: This method is unused

File: README.md [hunk 2]
```diff
@@ -54,7 +54,7 @@ When a specific path is configured in the settings, it takes precedence over the
 | PNPM | `pnpm` | `package.json` |
 | Yarn | `yarn` | `package.json` |
 | Golang | `go` | `go.mod` |
-| Python | `python3/pip3` or `python/pip` | `requirements.txt` |
+| Python | `python3/pip3` or `python/pip` | `requirements.txt`, `pyproject.toml` |
```
Comment (ruromero) [inline README.md]: Just a thought: We should consider using Pip/Pip3 instead of Python to be more accurate. Maybe the best moment is when we add support for uv/poetry.

**Review by ruromero — CHANGES_REQUESTED** [review]

File: src/providers/pyproject.toml.ts (hunk 1 above)
Comment (Strum355) [reply src/providers/pyproject.toml.ts]: [sdlc-workflow/verify-pr] Classified as **code change request** — sub-task [TC-4142](https://redhat.atlassian.net/browse/TC-4142) created to address this feedback.

File: package.json [hunk 3]
```diff
@@ -499,7 +525,7 @@
   "dependencies": {
     "@redhat-developer/vscode-redhat-telemetry": "^0.8.0",
     "@trustify-da/trustify-da-api-model": "^2.0.7",
-    "@trustify-da/trustify-da-javascript-client": "0.3.0-ea.b8af0f8",
+    "@trustify-da/trustify-da-javascript-client": "^0.3.0-ea.dcb2120",
```
[the remark this answers is not in the document]
Comment (Strum355) [reply package.json:528]: [sdlc-workflow/verify-pr] Classified as **suggestion** — the JS client version was explicitly requested by the PR author. The VS Code engine version compatibility is a pre-existing concern not scoped to this task. No sub-task created.

File: README.md (hunk 2 above)
Comment (Strum355) [reply README.md]: [sdlc-workflow/verify-pr] Classified as **suggestion** — "Just a thought" / "We should consider" language indicates a future improvement proposal, not a required change for this PR. No sub-task created.

**Comment by Strum355** [comment]
## Verification Report for TC-3852 (commit 6358e1b)

| Check | Result | Details |
|-------|--------|---------|
| Review Feedback | WARN | 3 threads classified: 1 code change request (unused method → [TC-4142](https://redhat.atlassian.net/browse/TC-4142)), 2 suggestions (no action) |
| Root-Cause Investigation | DONE | implement-task phase: speculative helper committed without verifying references → [TC-4143](https://redhat.atlassian.net/browse/TC-4143) |
| Scope Containment | WARN | 5 out-of-scope files: package-lock.json (expected), config.ts (uv/poetry settings), diagnostics.ts, exhortServices.ts, tsconfig.json |
| Diff Size | PASS | 950+/38- across 11 files — proportional to new provider + tests + lock file |
| Commit Traceability | PASS | Commit body references "Implements TC-3852" |
| Sensitive Patterns | PASS | Matches are false positives: `SecretStorage` type, `process.env` assignments |
| CI Status | PASS | build-and-test ✓, main ✓ |
| Acceptance Criteria | PASS | 7/7 criteria met |
| Test Quality | PASS | 13 tests with doc comments, no repetitive patterns |
| Verification Commands | N/A | No verification commands in task |

### Overall: WARN

Review feedback sub-task [TC-4142](https://redhat.atlassian.net/browse/TC-4142) created for unused `keyPathStartsWith` method removal. 5 out-of-scope files present but all related to pyproject.toml support (config for uv/poetry binary paths, diagnostics/exhort registration, tsconfig adjustments).

---
*This comment was AI-generated by [sdlc-workflow/verify-pr](https://github.com/mrizzi/sdlc-plugins) v0.5.11.*

[TC-4142]: https://redhat.atlassian.net/browse/TC-4142?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[TC-4143]: https://redhat.atlassian.net/browse/TC-4143?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[TC-4142]: https://redhat.atlassian.net/browse/TC-4142?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Review by ruromero — APPROVED** [review]
